tests/lib/tools/snapd-state

```diff
@@ -56,17 +56,17 @@ change_snap_channel() {
         echo "snapd-state: snap and channel are required parameters"
         exit 1
     fi
-    jq ".data.snaps[\"$SNAP\"].channel = \"$CHANNEL\"" /var/lib/snapd/state.json > /var/lib/snapd/state.json.new
+    jq ".data.snaps[\"$SNAP\"].channel = \"$CHANNEL\"" < /var/lib/snapd/state.json > /var/lib/snapd/state.json.new
```

Comment:
I am surprised this works as stdin redirections are still open files that apparmor re-validates on profile change.

Reply:
but they are opened by the open shell then piped into jq's stdin, the shell on core is not confined

Reply:
Should apparmor be checking correct access through mapping file descriptors to file descriptions and checking if the process has access the file description source? if that's the case then that wouldn't be very practical. I imagine the idea was having a more privileged process do needed setup, like snap-confine does and then hand over to less privileged process

Reply:
I think I figured this out and yes it does help for tests running as root.

Reply:
Thank you for looking into this